tests/variant_test.go

```diff
@@ -0,0 +1,247 @@
+// Licensed to ClickHouse, Inc. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. ClickHouse, Inc. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/chcol"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+var variantTestDate, _ = time.Parse(time.RFC3339, "2024-12-13T02:09:30.123Z")
+
+func setupVariantTest(t *testing.T) driver.Conn {
+	conn, err := GetNativeConnection(clickhouse.Settings{
+		"max_execution_time":              60,
+		"allow_experimental_variant_type": true,
+	}, nil, &clickhouse.Compression{
+		Method: clickhouse.CompressionLZ4,
+	})
+	require.NoError(t, err)
+
+	if !CheckMinServerServerVersion(conn, 24, 1, 0) {
+		t.Skip(fmt.Errorf("unsupported clickhouse version for Variant type"))
+		return nil
+	}
+
+	return conn
+}
+
+func TestVariant(t *testing.T) {
```

Comment:
Is there a need to test Variant:

- stdlib driver?
- with HTTP protocol? Do you expect any differences?